src/xiaomi_ble/parser.py

```diff
@@ -1461,6 +1470,140 @@ def obj4e1c(
     return {"device reset": True}
 
 
+def obj5003(
+    xobj: bytes, device: XiaomiBluetoothDeviceData, device_type: str
+) -> dict[str, Any]:
+    """Battery"""
+    device.update_predefined_sensor(SensorLibrary.BATTERY__PERCENTAGE, xobj[0])
+    return {}
+
+
+def obj5414(
+    xobj: bytes, device: XiaomiBluetoothDeviceData, device_type: str
+) -> dict[str, Any]:
+    """Device mode (KSI and KSIBP, not used in HA)"""
+    return {"mode": xobj[0]}
+
+
+def obj560c(
+    xobj: bytes, device: XiaomiBluetoothDeviceData, device_type: str
+) -> dict[str, Any]:
+    """Button press"""
+    if device_type in ["KS1", "KS1BP"]:
+        # button click 1-4
+        button = xobj[0]
+        if button == 1:
+            # left button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_left",
+                event_type="press",
+                event_properties=None,
+            )
+        if button == 2:
+            # mid left button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_mid_left",
+                event_type="press",
+                event_properties=None,
+            )
+        if button == 3:
+            # mid right button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_mid_right",
+                event_type="press",
+                event_properties=None,
+            )
+        if button == 4:
+            # right button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_right",
+                event_type="press",
+                event_properties=None,
+            )
+        else:
+            return {}
+    return {}
+
+
+def obj560d(
+    xobj: bytes, device: XiaomiBluetoothDeviceData, device_type: str
+) -> dict[str, Any]:
+    """Double button press"""
+    if device_type in ["KS1", "KS1BP"]:
```

Comment:
condition could be reversed to return early if its not a matching device type.
```suggestion
    if device_type not in ["KS1", "KS1BP"]:
       return {}
```